rfcs/syntax-string-interpolation.md

```diff
@@ -0,0 +1,122 @@
+# String interpolation
+
+## Summary
+
+New string interpolation syntax, and extending `string.format` to accept any arbitrary values without being exact about its types.
+
+## Motivation
+
+The problems with `string.format` are many.
+
+1. Must be exact about the types and its corresponding value.
+2. Using `%d` is the idiomatic default for most people, but this loses precision.
+   * `%d` casts the number into `long long`, which has a lower max value than `double` and does not support decimals.
+   * `%f` by default will format to the millionths, e.g. `5.5` is `5.500000`.
+   * `%g` by default will format up to the hundred thousandths, e.g. `5.5` is `5.5` and `5.5312389` is `5.53123`. It will also convert the number to scientific notation when it encounters a number equal to or greater than 10^6.
+   * To not lose precision, you need to use `%s`, but even so the type checker assumes you actually wanted strings.
+3. No support for `boolean`. You must use `%s` **and** call `tostring`.
+4. No support for values implementing the `__tostring` metamethod.
```

Comment:
The problem with `%s` calling `tostring` here is that it would cause more program to be a lot more lenient. I'd prefer to be more explicit, hence a different format syntax. I'll add this to alternatives to acknowledge this option.

`%q` is a no-go because it adds escaping backslashes before double quotes for strings, which is likely undesirable.